sympy/solvers/ode/single.py

```diff
@@ -2768,6 +2768,126 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
             + C2*bessely(n/d4,a4*x**d4/d4))).subs(x, x-b4))]
 
+class SecondLinearBesselSymbolic(SingleODESolver):
+    r"""
+    Solves second-order ODEs of the form y'' + A*x^k*y = 0 where k is symbolic.
+
+    These equations have solutions in terms of Bessel functions.
+
+    The general solution is:
+
+    .. math:: y(x) = \sqrt{x} \left[C_1 J_\nu(z) + C_2 Y_\nu(z)\right]
+
+    where :math:`\nu = \frac{1}{k+2}` and :math:`z = \frac{2}{k+2} x^{(k+2)/2}`
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, Function, dsolve, Eq, Derivative
+    >>> x = symbols('x')
+    >>> k = symbols('k')
+    >>> y = Function('y')
+    >>> ode = Eq(x**k*y(x) + Derivative(y(x), (x, 2)), 0)
+    >>> dsolve(ode)
+    Eq(y(x), sqrt(x)*(C1*besselj(1/(k + 2), 2*x**(k/2 + 1)/(k + 2)) + C2*bessely(1/(k + 2), 2*x**(k/2 + 1)/(k + 2))))
+
+    References
+    ==========
+    - Abramowitz & Stegun, Handbook of Mathematical Functions, 9.1.52
+    - https://eqworld.ipmnet.ru/en/solutions/ode/ode0201.pdf
+    """
+    hint = "2nd_linear_bessel_symbolic"
+    has_integral = False
+    order = [2]
+
+    def _matches(self):
+        eq = self.ode_problem.eq_high_order_free
+        f = self.ode_problem.func
+        x = self.ode_problem.sym
+        order = self.ode_problem.order
+
+        if order != 2:
+            return False
+
+        # Get linear coefficients
+        match = self.ode_problem.get_linear_coefficients(eq, f, order)
+        if not match:
+            return False
+
+        # We need: a3*y'' + b3*y' + c3*y = 0
+        # For Bessel with symbolic exponent: y'' + 0*y' + (x^k)*y = 0
+        a3 = match[2]  # coefficient of y''
+        b3 = match[1]  # coefficient of y'
+        c3 = match[0]  # coefficient of y
+
+        # Check if b3 = 0 (no first derivative term)
+        if b3 != 0:
+            return False
+
+        # Normalize: divide by coefficient of y''
+        if a3 == 0:
+            return False
+
+        if a3 != 1 and a3 != S.One:
+            if not a3.is_constant():
+                return False
+            c3 = c3 / a3
+
+        # Check if c3 is of the form A*x^k where k has free symbols
+        # This indicates symbolic exponent
+        A_sym = Wild('A_sym', exclude=[x])
+        k_sym = Wild('k_sym', exclude=[x])
+
+        pattern_match = c3.match(A_sym * x**k_sym)
+        if not pattern_match:
+            pattern_match = c3.match(x**k_sym)
+            if pattern_match:
+                pattern_match[A_sym] = S.One
+
+        # Store the matched values
+        if not pattern_match or k_sym not in pattern_match:
+            return False
+
+        # Check if k is symbolic (has free symbols)
+        k_val = pattern_match[k_sym]
+
+        # Exclude k=0 and k=1 cases (handled by other solvers)
+        # k=0: constant coefficient ODEs
+        # k=1: Airy equation (handled by 2nd_linear_airy)
+        if k_val.is_number and k_val in [0, 1]:
+            return False
```

Comment:
As far as I can tell the solutions are valid for `k = 0` and `k = 1` and at least for most real values of `k` I have checked with the notable exception being `k = -2` which is not checked here.

The fact that other solvers can handle this is not a reason for not returning a valid answer here. It just means that the other solvers should be preferred by the solver ordering.